plugin-server/src/ingestion/ingestion-consumer.ts

```diff
@@ -528,29 +537,54 @@ export class IngestionConsumer {
                 continue
             }
 
-            const eventKey = `${event.token}:${event.distinct_id}`
-
             if (this.shouldSkipPerson(event.token, event.distinct_id)) {
                 event.properties = {
                     ...(event.properties ?? {}),
                     $process_person_profile: false,
                 }
             }
 
+            batch.push({ message, event })
+        }
+
+        return Promise.resolve(batch)
+    }
+
+    private groupEventsByDistinctId(messages: IncomingEventWithTeam[]) {
+        const batches: IncomingEventsByDistinctId = {}
+        for (const { event, message, team, token } of messages) {
+            const eventKey = `${token ?? ''}:${event.distinct_id}`
```

Comment:
Now we have `team` here we could add `team.id` to the `eventKey` if we wanted to